lib/ssf/client.rb

```diff
@@ -48,18 +49,25 @@ def start_span(name: '', tags: {}, parent: nil, indicator: false)
           parent_id: parent.id,
           indicator: indicator,
           clean_tags: false,
+          override_service: override_service
         )
       else
         start_span_from_context(
           name: name,
           tags: tags,
           indicator: indicator,
           clean_tags: false,
+          override_service: override_service
         )
       end
     end
 
-    def start_span_from_context(name: '', tags: {}, trace_id: nil, parent_id: nil, indicator: false, clean_tags: true)
+    def start_span_from_context(name: '', tags: {}, trace_id: nil, parent_id: nil, indicator: false, clean_tags: true, override_service: nil)
+      span_service = override_service || @service
+      unless span_service
```

Comment:
Hm, I think we might just prohibit service-less clients from being created and save ourselves this check. What do you think - do we have things with clients like that out there?